app/models/user.rb

```diff
@@ -18,10 +18,26 @@ def stats
     total_prayers = Prayer.where(user_id: id).count
     total_duration = routes.sum(:seconds)
 
+    current_commitment = self.current_commitment
+    current_journey = current_commitment&.journey
+    if current_commitment
+      commitment_distance = routes.where(commitment_id: current_commitment.id).sum(:mileage)
+      commitment_duration = routes.where(commitment_id: current_commitment.id).sum(:seconds)
+      commitment_prayers = Prayer.joins(:route).where(routes: { commitment_id: current_commitment.id }).count
```

Comment:
Nice!

Similar to my other comment, this isn't wrong, but it can also be achieved through ActiveRecord associations.
A Commitment could have many Routes, and a Route could have many Prayers, then you can set up Commitment to have many Prayers through Routes, which would let you call `current_commitment.prayers.count`.
I'm pretty sure it would construct the same query you've got here.